server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -234,21 +240,74 @@ static SecureSettings loadSecureSettings(Environment initialEnv) throws Bootstra
             throw new BootstrapException(e);
         }
 
+        char[] passChars;
         try {
+            if (keystore != null && keystore.hasPassword()) {
+                passChars = readPassphrase(System.in, KeyStoreAwareCommand.MAX_PASSPHRASE_LENGTH);
+            } else {
+                passChars = new char[0];
+            }
+        } catch (IOException e) {
+            throw new BootstrapException(e);
+        }
+
+        try (SecureString password = new SecureString(passChars)) {
```

Comment:
future thought (doesn't need to be in this PR): maybe readPassphrase should return a SecureString.